assets/robotest/Makefile

```diff
@@ -0,0 +1,306 @@
+# is expected to be overridden (to avoid wipe on `make clean`).
+ROBOTEST_CACHE_ROOT ?= $(BUILDDIR)/cache
+# ROBOTEST_CACHEDIR is the flavor specific cache directory.
+# If the cache is in the builddir, it is already enterprise/oss specific. If not
+# we need to disambiguate between the two.
+ifeq ($(BUILDDIR), $(realpath $(dir $(ROBOTEST_CACHE_ROOT))))
+# ROBOTEST_CACHEDIR is the flavor specific cache directory.
+ROBOTEST_CACHEDIR ?= $(ROBOTEST_CACHE_ROOT)
+else
+ROBOTEST_CACHEDIR ?= $(ROBOTEST_CACHE_ROOT)/$(ROBOTEST_CACHE_FLAVOR)
+endif
+# ROBOTEST_GRAVITY_PKG_CACHE is a gravity state dir storing packages
+# downloaded from the default distribution infrastructure.
+# ROBOTEST_GRAVITY_PKG_CACHE may be shared between many parallel robotest builds
+# in CI infra. A reasonable override for personal use is sharing with:
+#   $(HOME)/.gravity
+ROBOTEST_GRAVITY_PKG_CACHE ?=  $(ROBOTEST_CACHEDIR)/gravity-state
+# ROBOTEST_CACHE_TMP is a staging directory for building/copying/unpacking content
+# before an atomic rename into the. Must be on the same filesystem as ROBOTEST_TELE_CACHE
+ROBOTEST_CACHE_TMP = $(ROBOTEST_CACHEDIR)/tmp
+# ROBOTEST_TELE_CACHE is a filesytem cache storing prebuilt tele binaries for
+# different gravity versions. ROBOTEST_TELE_CACHE may be shared between many
+# parallel robotest builds in CI infra.
+ROBOTEST_TELE_CACHE ?= $(ROBOTEST_CACHEDIR)/tele
+# ROBOTEST_CACHES are all the directories on this machine used to speed up
+# upgrade artifact contruction by avoiding redownloading/rebuilding released
+# artifacts.
+ROBOTEST_CACHES = $(ROBOTEST_GRAVITY_PKG_CACHE) $(ROBOTEST_TELE_CACHE)
+
+# Image configuration.
+
+# ROBOTEST_IMAGEDIR is a store of robotest images used only for the current
+# robotest run. Expected to be populated on demand from the caches and mounted
+# as a volume into the robotest container. This exists to prevent any
+# concurrency issues stemming from using cached images directly (e.g. if the
+# cache is edited/cleared during a test run).
+ROBOTEST_IMAGEDIR = $(ROBOTEST_BUILDDIR)/images
+
+# ROBOTEST_IMAGE_SRCDIR is the directory containing all sources to build the robotest application.
+# It is mounted into a build container.
+ROBOTEST_IMAGE_SRCDIR = $(TOP)/current
+# ROBOTEST_IMAGE_SRC are the files that affect the robotest application.  If any
+# of these change, all robotest images must be rebuilt.
+ROBOTEST_IMAGE_SRC := $(shell find $(ROBOTEST_IMAGE_SRCDIR) -type f)
+# ROBOTEST_IMAGE_MANIFEST is the gravity application manifest to build into robotest images.
+ROBOTEST_IMAGE_MANIFEST = $(ROBOTEST_IMAGE_SRCDIR)/app.yaml
+
+# ROBOTEST_UPGRADE_BASE_IMAGE_SRCDIR is the directory containing all sources to build the
+# robotest upgrade base application. It is mounted into a build container.
+ROBOTEST_UPGRADE_BASE_IMAGE_SRCDIR = $(TOP)/upgrade-base
+# ROBOTEST_UPGRADE_BASE_IMAGE_SRC is the gravity application manifest to build into robotest images.
```

Comment:
from the implementation it looks as a list of files in the `$ROBOTEST_UPGRADE_BASE_IMAGE_SRCDIR` directory - copy/paste comment?

Reply:
Fixed in d2cfbf4.